@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -1441,22 +1441,27 @@ export class TextElementsRenderer {
     }
 
     private addPointLabel(
-        pointLabel: TextElement,
         labelState: TextElementState,
         groupState: TextElementGroupState,
         position: THREE.Vector3,
         screenPosition: THREE.Vector2,
         poiRenderer: PoiRenderer,
         textCanvas: TextCanvas,
         renderParams: RenderParams,
-        temp: TempParams
+        temp: TempParams,
+        iconIndex?: number
     ): boolean {
-        const textRenderState: RenderState = labelState.textRenderState!;
-        const iconRenderState: RenderState = labelState.iconRenderState!;
+        const pointLabel: TextElement = labelState.element;
+        const textRenderState: RenderState | undefined = labelState.textRenderState;
+        const iconRenderState: RenderState =
+            iconIndex !== undefined
+                ? labelState.iconRenderStates![iconIndex]
+                : labelState.iconRenderState!;
         const poiTextMaxDistance = getMaxViewDistance(
             this.m_mapView,
             this.m_options.maxDistanceRatioForPoiLabels!
         );
+        const hasText = textRenderState !== undefined && pointLabel.text !== "";
```

Comment:
move closer to its usage

Reply:
Done. I couldn't move it much closer though, since one usage is deeply nested in 5 levels of conditional blocks.

Reply:
Most of this code is changing in the follow-up review I mentioned, so no much use in changing this here.